### PR TITLE
Named MPS - fix duplicated "ranged" binding constraints

### LINKED ISSUE #1562: Binding constraints : duplicate lines in named MPS files
See https://github.com/AntaresSimulatorTeam/SimTest/releases/download/v8.7.0/valid-named-mps-ubuntu-20.04.zip, file binding-constraints/output/reference/problem-1-1--optim-nb-1.mps
```
    NTCDirect::link<sink-7$$source-7>::hour<63>  H2::hourly::hour<61>  1.0000000000
    NTCDirect::link<sink-7$$source-7>::hour<63>  H2::hourly::hour<61>  1.0000000000
```
Relevant extract from binding-constraints/input/bindingconstraints/bindingconstraints.ini
```
[5]
name = H2
id = h2
enabled = true
type = hourly
operator = both
sink-7%source-7 = 1.000000%2
sink-8%source-8 = -1.000000%1
```
I suspect that the offset (after %) is not handled properly.

### PR DESCRIPTION
**Pull request by a-zakir**
## TODO
- [ ] Fix CI (re-generate reference results)
- [x] Check impacts on infeasible problem analyzer
- [x] it broke theses tests (non-exhaustive for the moment):
    -  'named  & anonymous mps' tests
    -  'test_constraint' part of 'unit and end-to-end' 
    - short-tests /
                      053 System Map Editor - 1
                      061 Four areas - Grid outages  05
                      055 System Map Editor - 3
                     060 Four areas - Grid outages  04
   - medium-tests /
                            077 KCG on regional dataset 02
                            078 KCG on regional dataset 03
   
see [Antares_Simulator_Tests_NR/pull/9](https://github.com/AntaresSimulatorTeam/Antares_Simulator_Tests_NR/pull/9)
#close #1562

### CODE REVIEW

File: src/libs/antares/study/binding_constraint/BindingConstraintsRepository.cpp [hunk 1]
```diff
@@ -73,11 +73,48 @@ static int valueForSort(BindingConstraint::Operator op)
...
+        std::string::iterator left = tmp_str1.begin() + left_pos;
+        size_t right_pos = 0;
+        if (right_pos = tmp_str2.find_last_of("_gt"); right_pos == std::string::npos)
+        {
+            logs.error() << "Ranged constraints comparison : " << tmp_str1 << " and " << tmp_str2
+                         << " has failed!";
+        }
+        std::string::iterator right = tmp_str2.begin() + right_pos;
+
+        auto new_s1 = std::string(tmp_str1.begin(), left - 2);
```
Comment (JasonMarechal25) [inline src/libs/antares/study/binding_constraint/BindingConstraintsRepository.cpp]: Why not use use:
`auto new_s1 = tmp_str1.substr(0, left_pos);`

**Review by JasonMarechal25 — COMMENTED** [review]
I see you added "_gt" and "_lt" suffixes to constraints name in Loader. Does it create any change in MPS produced?

**Comment by a-zakir** [comment]
> I see you added "_gt" and "_lt" suffixes to constraints name in Loader. Does it create any change in MPS produced?

absolutely! instead of  having 2 constraints with the same name.

File: src/libs/antares/study/binding_constraint/BindingConstraintsRepository.cpp (hunk 1 above)
Comment (a-zakir) [reply src/libs/antares/study/binding_constraint/BindingConstraintsRepository.cpp]: thanks

**Comment by JasonMarechal25** [comment]
> > I see you added "_gt" and "_lt" suffixes to constraints name in Loader. Does it create any change in MPS produced?
> 
> absolutely! instead of having 2 constraints with the same name.

Just naming, no difference of results? Any difference in the order of apparition of constraints in the mps files?

**Comment by a-zakir** [comment]
> > > I see you added "_gt" and "_lt" suffixes to constraints name in Loader. Does it create any change in MPS produced?
> > 
> > 
> > absolutely! instead of having 2 constraints with the same name.
> 
> Just naming, no difference of results? Any difference in the order of apparition of constraints in the mps files?

that's  what i was saying in the daily for days, in named constraints ('gt' and 'lt'), the order matters!

**Comment by a-zakir** [comment]
> > > > I see you added "_gt" and "_lt" suffixes to constraints name in Loader. Does it create any change in MPS produced?
> > > 
> > > 
> > > absolutely! instead of having 2 constraints with the same name.
> > 
> > 
> > Just naming, no difference of results? Any difference in the order of apparition of constraints in the mps files?
> 
> that's what i was saying in the daily for days, in named constraints ('gt' and 'lt'), the order matters!

see broken tests list

**Review by flomnes — COMMENTED** [review]
Why is it necessary to take `_lt`/`_gt` suffixes into account when comparing binding constraints by name ?

**Comment by a-zakir** [comment]
> Why is it necessary to take `_lt`/`_gt` suffixes into account when comparing binding constraints by name ?

without this dev the comparison   ret = ConstraintName("Bc_lt)" < ConstraintName("Bc_gt")  = false, 
in order to keep the same solution we must add  the "_lt" one before the "_gt", thus ret has to be true.

also ConstraintName("BcNOT_FROM_THE-SAME_RANGED_lt)" < ConstraintName("Bc_gt")  =false

**Comment by flomnes** [comment]
LGTM @a-zakir Please revert YML files.

**Review by flomnes — APPROVED** [review]

**Pull request merged**